src/broker/amqp.rs

```diff
@@ -213,11 +213,9 @@ impl Broker for AMQPBroker {
     }
 
     async fn ack(&self, delivery: &Self::Delivery) -> Result<(), BrokerError> {
-        let _lock = self.consume_channel_write_lock.lock().await;
```

Comment:
Did you mean to take this out? If so, are you sure it's okay to not acquire this lock here?

https://github.com/rusty-celery/rusty-celery/blob/ddebcb3258f53e584b981edc3824062f9d6d6e12/src/broker/amqp.rs#L134-L140

Reply:
I'm not sure what the lock was for in the first place?

- the acker guarantees that you use the same channel for acking
- you can ack at any time, even if the channel is used by another thread, whether to read or write data, lapin ensures the correct ordrering of frames

Reply:
Your 2nd bullet was what my question was really about.

> lapin ensures the correct ordering of frames

Is that a somewhat recent change? I originally added the lock because the worker was crashing when running on multiple threads under load. The error I was getting suggested it was due to incorrect frame ordering.

Reply:
There used to be a problem wrt basic_publish frame ordering when other frames were in flight because the behaviour was poorly documented. We now properly consider all the frames from a basic publish to be contiguous as expected by rabbitmq.
The fix is fairly recent indeed (september) as it took some time to track down (actually had to read rabbitmq server code)

Reply:
1.2.6 and 1.2.7 were the relevant bugfix releases

Reply:
Ahh gotcha! So in that case I think we can remove the `consume_channel_write_lock`:

https://github.com/rusty-celery/rusty-celery/blob/ddebcb3258f53e584b981edc3824062f9d6d6e12/src/broker/amqp.rs#L141

And we can probably use a `RwLock` instead of a `Mutex` around the `produce_channel`, right?

https://github.com/rusty-celery/rusty-celery/blob/ddebcb3258f53e584b981edc3824062f9d6d6e12/src/broker/amqp.rs#L132

I could make these changes in a separate PR unless you want to tack that onto this PR.

> actually had to read rabbitmq server code

That sounds like fun 😝